test/soft_delete_repo_test.exs

```diff
@@ -90,6 +90,55 @@ defmodule Ecto.SoftDelete.Repo.Test do
     end
   end
 
+  describe "soft_restore/1" do
+    test "should soft restore the queryable" do
+      user = Repo.insert!(%User{email: "test0@example.com"})
+      user = Repo.soft_delete!(user)
+      Repo.soft_restore(user, Repo)
+
+      assert Repo.get_by!(User, [email: "test0@example.com"], with_deleted: true).deleted_at ==
+               nil
+    end
+  end
+
+  describe "soft_restore_all/1" do
+    test "soft deleted the query" do
+      Repo.insert!(%User{email: "test0@example.com"})
+      Repo.insert!(%User{email: "test1@example.com"})
+      Repo.insert!(%User{email: "test2@example.com"})
+
+      assert Repo.soft_delete_all(User) == {3, nil}
+
+      assert User |> Repo.all(with_deleted: true) |> length() == 3
+
+      assert %DateTime{} =
+               Repo.get_by!(User, [email: "test0@example.com"], with_deleted: true).deleted_at
+
+      assert %DateTime{} =
+               Repo.get_by!(User, [email: "test1@example.com"], with_deleted: true).deleted_at
+
+      assert %DateTime{} =
+               Repo.get_by!(User, [email: "test2@example.com"], with_deleted: true).deleted_at
+
+      assert Repo.soft_restore_all(%User{}, Repo) == {3, nil}
+
+      assert User |> Repo.all(with_deleted: true) |> length() == 3
+
+      assert Repo.get_by!(User, [email: "test0@example.com"], with_deleted: true).deleted_at ==
+               nil
+
+      assert Repo.get_by!(User, [email: "test1@example.com"], with_deleted: true).deleted_at ==
+               nil
+
+      assert Repo.get_by!(User, [email: "test2@example.com"], with_deleted: true).deleted_at ==
+               nil
+    end
+
+    test "when no results are found" do
+      assert Repo.soft_delete_all(User) == {0, nil}
+    end
+  end
```

Comment:
I know you copy / past from a previous test, but IMO thoses tests are not really interesting, they check that SQL works. If there is an issue it will not be related to this lib.

Maybe just check the ecto request could be enough